backend/employee-payroll-service/src/main/java/com/mycompany/employee/payroll/service/controller/EmployeeController.java

```diff
@@ -0,0 +1,31 @@
+package com.mycompany.employee.payroll.service.controller;
+
+import com.mycompany.employee.payroll.service.dto.EmployeeDto;
+import com.mycompany.employee.payroll.service.service.EmployeeService;
+import com.mycompany.employee.payroll.service.vo.EmployeeVo;
+import org.hibernate.service.spi.ServiceException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/employee")
+public class EmployeeController {
+
+  private final EmployeeService employeeService;
+
+  public EmployeeController(EmployeeService employeeService) {
+    this.employeeService = employeeService;
+  }
+
+  @PostMapping("/create")
+  public ResponseEntity<EmployeeVo> create(@Validated @RequestBody EmployeeDto dto)
+      throws ServiceException {
+    return new ResponseEntity<>(employeeService.create(dto), HttpStatus.CREATED);
```

Comment:
Technically, this short hand return is working fine. However, in the future, for PR, let's separate service data and response entity.

Reply:
Sorry for not giving the details clearly.

What I meant here is

```java
public ResponseEntity<EmployeeVo> create(@Validated @RequestBody EmployeeDto dto)
      throws ServiceException {
    EmployeeVo emp = employeeService.create(dto);
    // We can do whatever we like e.g. logging
    return new ResponseEntity<>(emp, HttpStatus.CREATED);
}
```